src/lib.rs

```diff
@@ -496,6 +548,11 @@ impl Header {
     /// file.
     pub fn set_metadata(&mut self, meta: &fs::Metadata) {
         self.fill_from(meta);
+        // Set size of directories to zero
+        let htype = self.link[0];
+        self.set_size(if htype == b'5' { 0 } else { meta.len() });
```

Comment:
Perhaps this could use `meta.is_dir()` instead of checking for `b'5'` (somewhat obscure)